src/tokens/slice.ts

```diff
@@ -149,19 +149,30 @@ const slice = createSlice({
       loading: false,
       error: true,
     }),
-    importToken: (state, action: PayloadAction<BaseToken>) => ({
-      ...state,
-      importedTokens: {
-        ...state.importedTokens,
-        [action.payload.tokenId]: {
-          ...action.payload,
-          // Force imported tokens to be visible even with zero balance.
-          showZeroBalance: true,
-          balance: null,
-          isManuallyImported: true,
+    importToken: (state, action: PayloadAction<StoredTokenBalance>) => {
+      const importedTokenDetails = {
+        ...action.payload,
+        // Force imported tokens to be visible even with zero balance.
+        showZeroBalance: true,
+        isManuallyImported: true,
+      }
+
+      return {
+        ...state,
+        tokenBalances: {
+          ...state.tokenBalances,
+          [action.payload.tokenId]: importedTokenDetails,
         },
-      },
-    }),
+        importedTokens: {
+          ...state.importedTokens,
+          [action.payload.tokenId]: {
+            ...importedTokenDetails,
```

Comment:
why are we storing the imported token details in both `importedTokens` and `tokenBalances`? we should really only have one source of truth in redux, perhaps the importedTokens state only needs to have a list of id's if you're relying on `tokenBalances` for the token info?

Reply:
I interpret these fields as:
`tokenBalances` the source of truth for showing tokens with balances.
`importedTokens` the source of truth for the static data of the imported tokens.

But I understand we could derive `importedTokens` from `tokenBalances`, but somehow I feel it changes the meaning of `tokenBalances`, because now it's not only showing tokens with balance but also stores information about the imported tokens. But maybe we are fine we that change.

Reply:
I made this change [here](https://github.com/valora-inc/wallet/pull/4798/commits/84cb2708bfe70debe00c50a8e949ea65a2b3e032).

I think you were right, we can live without the `importedTokens` field.